samcli/commands/validate/lib/sam_template_validator.py

```diff
@@ -120,6 +121,23 @@ def _replace_local_codeuri(self):
                 if "DefinitionUri" in resource_dict:
                     SamTemplateValidator._update_to_s3_uri("DefinitionUri", resource_dict)
 
+    def _replace_local_image(self):
+        """
+        Adds fake ImageUri to AWS::Serverless::Functions that reference a local image using Metadata.
+        This ensures sam validate works without having to package the app or use ImageUri.
+        """
+        resources = self.sam_template.get("Resources", {})
+        for _, resource in resources.items():
+            resource_type = resource.get("Type")
+            properties = resource.get("Properties", {})
+
+            is_image_function = resource_type == "AWS::Serverless::Function" and properties.get("PackageType") == IMAGE
```

Comment:
should this be an enum we can look at? Otherwise this looks great 🎉

Reply:
`AWS_SERVERLESS_FUNCTION` from samcli/commands/_utils/resources.py:8